lib/dockerregistry/inventory.go

```diff
@@ -1973,6 +1976,60 @@ func (sc *SyncContext) RunChecks(
 	return nil
 }
 
+// Run is a function of ImageSizeCheck and checks that all
+// images to be promoted are under the max file size.
+func (check *ImageSizeCheck) Run() error {
+	const mbToByteShift = 20
+	maxImageSizeByte := check.MaxImageSize << mbToByteShift
+	oversizedImages := make([]string, 0)
+	invalidImages := make([]string, 0)
+	for edge := range check.PullEdges {
+		imageSize := check.DigestImageSize[edge.Digest]
+		if imageSize > maxImageSizeByte {
+			oversizedImages = append(oversizedImages,
+				string(edge.DstImageTag.ImageName))
+		}
+		if imageSize <= 0 {
+			invalidImages = append(invalidImages,
+				string(edge.DstImageTag.ImageName))
+		}
+	}
+
+	errStr := ""
```

Comment:
You can just use fmt.Errorf() directly and return it when you use it, instead of appending into this errStr.

Reply:
I'm using an errStr so that I can bubble up errors for both oversized images and invalid images. if I use fmt.Errorf() directly how would I bubble up the two errors into one